addons/payment/views/payment_templates.xml

```diff
@@ -368,27 +368,50 @@
         -->
-        <div t-if="tx.state == 'pending' and not is_html_empty(tx.acquirer_id.sudo().pending_msg)"
-             class="alert alert-warning alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().pending_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'authorized' and not is_html_empty(tx.acquirer_id.sudo().auth_msg)"
-             class="alert alert-success alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().auth_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'done' and not is_html_empty(tx.acquirer_id.sudo().done_msg)"
-             class="alert alert-success alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().done_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'cancel' and not is_html_empty(tx.acquirer_id.sudo().cancel_msg)"
-             class="alert alert-danger alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().cancel_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
+        <t t-if="tx.state == 'draft'">
+            <t t-set="alert_style">info</t>
+            <t t-set="status_message">
+                <p>Your payment has not been processed yet.</p>
+            </t>
+        </t>
+        <t t-elif="tx.state == 'pending'">
+            <t t-set="alert_style">warning</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().pending_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().pending_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'authorized'">
+            <t t-set="alert_style">success</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().auth_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().auth_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'done'">
+            <t t-set="alert_style">success</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().done_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().done_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'cancel'">
+            <t t-set="alert_style">danger</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().cancel_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().cancel_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'error'">
+            <t t-set="alert_style">danger</t>
+            <t t-set="status_message">
+                <p>An error occurred during the processing of your payment.</p>
+            </t>
+        </t>
+
+        <div t-if="status_message or tx.state_message"
```

Comment:
Might be simpler to check here the html emptiness (instead of 4 checks above) :thinking:

Reply:
I'm tempted, but it would actually require 2 checks instead of 1, and they would be slightly more out of context than the 4 above... so let's keep it like this.